### ISSUE TITLE
EPSGs Where Unit is FT

### ISSUE THREAD

**Issue by dusko23** [wfletcher/EasyGIS.NET#124]
The UnitsToMeters factor is 1 for all EPSG Codes.
 That doesn't look right.
I have tested using grid 6785 where the unit is ft.
When drawing two points that should be 10 meters apart, the real distance is about 2.8 ft which is 0.85344 meters.
Is there a cure for this in proj9 database? 

![image](https://github.com/wfletcher/EasyGIS.NET/assets/26789352/0c15f95a-5fae-4d86-b236-a74f01908469)

**Comment by wfletcher** [comment]
Yes currently it is 1 for all projected CRS as I have not figured out how to calculate the correct scaling factor. Until further investigation if you need to use this property you will need to calculate distances by transforming to geodetic coordinates

**Comment by dusko23** [comment]
I would suggest introducing another boolean property like "inMeters" which is TRUE if ft or other non-meter unit is used.
This would allow to calculate distance differently. 
I have a function in place to calculate distance if EPSG = pseudo Mercator.
Adding non-meter units will be easy.
This is my current function.

![image](https://github.com/wfletcher/EasyGIS.NET/assets/26789352/eb8e70e0-d026-4c54-9e33-93f99f2c38b1)

Using the library
https://coordinatesharp.com/

Calculating the distance on Ellipsoid (and sphere) using your library would be great :-)

**Comment by wfletcher** [comment]
I use the proj C API to perform all the CRS functionality and I have not been able to find functions to determine the units so I can;t add a property like inMeters.

If you want code to measure distances have a look at the code in the SFMap control.

```

private double DistanceBetweenPoints(PointD p0, PointD p1)
        {
            double distance = double.NaN;
            if ((this.MapCoordinateReferenceSystem as IGeographicCRS) != null)
            {
                //distance = EGIS.ShapeFileLib.ConversionFunctions.DistanceBetweenLatLongPointsHaversine(ConversionFunctions.Wgs84RefEllipse, p0.Y, p0.X, p1.Y, p1.X);
                distance = EGIS.ShapeFileLib.ConversionFunctions.GeodesicDistanceAndBearingBetweenLatLonPoints(ConversionFunctions.Wgs84RefEllipse, p0.Y, p0.X, p1.Y, p1.X).Item1;

            }
            else if ((this.MapCoordinateReferenceSystem as IProjectedCRS) != null)
            {
                using (var transformation = EGIS.Projections.CoordinateReferenceSystemFactory.Default.CreateCoordinateTrasformation(
                    this.MapCoordinateReferenceSystem,
                    EGIS.Projections.CoordinateReferenceSystemFactory.Default.GetCRSById(EGIS.Projections.CoordinateReferenceSystemFactory.Wgs84EpsgCode)))
                {
                    p0 = transformation.Transform(p0);
                    p1 = transformation.Transform(p1);
                    //distance = EGIS.ShapeFileLib.ConversionFunctions.DistanceBetweenLatLongPointsHaversine(ConversionFunctions.Wgs84RefEllipse, p0.Y, p0.X, p1.Y, p1.X);
                    distance = EGIS.ShapeFileLib.ConversionFunctions.GeodesicDistanceAndBearingBetweenLatLonPoints(ConversionFunctions.Wgs84RefEllipse, p0.Y, p0.X, p1.Y, p1.X).Item1;
                }
                //calculate euclidean distance
                //distance = Math.Sqrt((p1.X - p0.X) * (p1.X - p0.X) + (p1.Y - p0.Y) * (p1.Y - p0.Y));
            }
            return distance;
        }

```

**Comment by dusko23** [comment]
Thanks, will convert to VB and test.

**Comment by dusko23** [comment]
For the time being, I have implemented a solution based on my existing distance calculation.
If the second part of the EPSG description exposes (ft) string the global variable is set to true.
However, the user can change to METER and the global variable is set to false.
I assume the ProjectedCoordinateSystems EPSG codes are either meters or ft. 

![image](https://github.com/wfletcher/EasyGIS.NET/assets/26789352/f9f08d99-328b-417e-83e0-f38936eb04d3)

**Comment by wfletcher** [comment]
I would not make any assumptions about units being either meters or feet. They could be km, miles, yards, fathoms, furlongs or whatever the person that defined the CRS decided to use at the time it was created.
.
If you want to measure the distance between 2 points use the code I posted.

https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=&ved=2ahUKEwiUyNmyhLaCAxWCzzgGHegRDaMQFnoECDUQAQ&url=https%3A%2F%2Fdoc.arcgis.com%2Fen%2Fallsource%2Flatest%2Fvisualization%2Fpdf%2Fprojected_coordinate_systems.pdf&usg=AOvVaw0h3QaSOepZ6sTPuwp9TMN4&opi=89978449
